packages/react-reconciler/src/ReactFiberReconciler.js

```diff
@@ -368,6 +369,24 @@ if (__DEV__) {
     return copyWithSetImpl(obj, path, 0, value);
   };
 
+  // Support DevTools editable hooks state.
+  overrideHook = (
+    fiber: Fiber,
+    nativeHookIndex: number,
+    path: Array<string | number>,
+    value: any,
+  ) => {
+    let currentHook = fiber.memoizedState;
+    while (currentHook !== null && nativeHookIndex > 0) {
+      currentHook = currentHook.next;
+      nativeHookIndex--;
+    }
+    if (currentHook !== null) {
+      let updatedState = copyWithSet(currentHook.memoizedState, path, value);
+      currentHook.queue.dispatch(updatedState);
```

Comment:
This will work for `useState` but not `useReducer`, which I assume is intentional, correct?

Reply:
We really should support useReducer if we support useState since that’s the recommended one. Don’t want people choosing useState over useReducer based on DevTools.

Reply:
Interesting. I thought a little (not much) about `useReducer` support but didn't think it was necessary.